packages/embedder/indexing/retrospectiveDiscussionTopic.ts

```diff
@@ -0,0 +1,267 @@
+    await upsertEmbeddingsMetaRows(embeddingsMetaRows)
+  }
+  return
+}
+
+async function getPreferredNameByUserId(userId: string, dataLoader: DataLoaderWorker) {
+  const user = await dataLoader.get('users').load(userId)
+  return !user ? 'Unknown' : user.preferredName
+}
+
+async function formatThread(
+  dataLoader: DataLoaderWorker,
+  comments: Comment[],
+  parentId: string | null = null,
+  depth = 0
+): Promise<string> {
+  // Filter and sort comments as before
+  const filteredComments = comments
+    .filter((comment) => comment.threadParentId === parentId)
+    .sort((a, b) => a.threadSortOrder - b.threadSortOrder)
+
+  // Use map to create an array of promises for each formatted comment string
+  const formattedCommentsPromises = filteredComments.map(async (comment) => {
+    const indent = '   '.repeat(depth + 1)
+    const author = comment.isAnonymous
+      ? 'Anonymous'
+      : comment.createdBy
+      ? await getPreferredNameByUserId(comment.createdBy, dataLoader)
+      : 'Unknown'
+    const how = depth === 0 ? 'wrote' : 'replied'
+    const content = comment.plaintextContent
+    const formattedPost = `${indent}- ${author} ${how}, "${content}"\n`
+
+    // Recursively format child threads
+    const childThread = await formatThread(dataLoader, comments, comment.id, depth + 1)
+    return formattedPost + '\n' + childThread
+  })
+
+  // Resolve all promises and join the results
+  const formattedComments = await Promise.all(formattedCommentsPromises)
+  return formattedComments.join('')
+}
+
+export const createTextFromNewMeetingDiscussion = async (
+  newMeeting: MeetingRetrospective,
+  discussionId: string,
+  dataLoader: DataLoaderWorker,
+  textForReranking: boolean = false
+) => {
+  if (!newMeeting) throw 'newMeeting is undefined'
+  if (!isMeetingRetrospective(newMeeting)) throw 'newMeeting is not retrospective'
+  const template = await dataLoader.get('meetingTemplates').load(newMeeting.templateId)
+  if (!template) throw 'template is undefined'
+  const discussPhase = newMeeting.phases.find((phase) => phase.phaseType === 'discuss')
+  if (!discussPhase) throw 'newMeeting discuss phase is undefined'
+  if (!discussPhase.stages) throw 'newMeeting discuss phase has no stages'
+  const discussStage = discussPhase.stages.find(
+    (stage) => stage.id === discussionId
+  ) as DiscussStage
+  if (!discussStage) throw 'newMeeting discuss stage not found'
+  const {summary: discussionSummary} = (await pg
+    .selectFrom('Discussion')
+    .select('summary')
+    .where('id', '=', discussStage.discussionId)
+    .executeTakeFirst()) ?? {summary: null}
+  const r = await getRethink()
+  const reflectionGroup = await r
+    .table('RetroReflectionGroup')
+    .get(discussStage.reflectionGroupId)
+    .run()
+  const reflections = await r
+    .table('RetroReflection')
+    .getAll(reflectionGroup.id, {index: 'reflectionGroupId'})
+    .run()
+  const promptIds = [...new Set(reflections.map((r) => r.promptId!))]
```

Comment:
-1 `r.promptId` is guaranteed, no need for `!`
nit: even though it's in a different scope, calling things `r` in the presence of the rethinkdb `r` can be confusing